src/Analyzer.Utilities/Extensions/IOperationExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using static Analyzer.Utilities.DiagnosticHelpers;
+using Microsoft.CodeAnalysis;
+
+namespace Analyzer.Utilities
+{
+    public static class IOperationExtensions
+    {
+        public static bool HasConstantValue(this IOperation operation, string comparand, StringComparison comparison)
+        {
+            var constantValue = operation.ConstantValue;
+            if (!constantValue.HasValue)
+            {
+                return false;
+            }
+
+            if (operation.Type == null || operation.Type.SpecialType != SpecialType.System_String)
+            {
+                return false;
+            }
+
+            return string.Equals((string)constantValue.Value, comparand, comparison);
+        }
+
+        public static bool HasNullConstantValue(this IOperation operation)
+        {
+            return operation.ConstantValue.HasValue && operation.ConstantValue.Value == null;
+        }
+
+        public static bool HasConstantValue(this IOperation operation, long comparand)
+        {
+            return operation.HasConstantValue(unchecked((ulong)(comparand)));
+        }
+
+        public static bool HasConstantValue(this IOperation operation, ulong comparand)
```

Comment:
FYI @JohnHamby @genlu: Comparing `IOperation.ConstantValue` with primitive type constants doesn't seem to be trivial as the analyzer needs to handle all type kinds for the operation's constant value - lot of our analyzers don't do this correctly. We should consider exposing public APIs on IOperation for constant value comparison.